qiskit/circuit/classical/types/ordering.py

```diff
@@ -83,14 +83,39 @@ def order(left: Type, right: Type, /) -> Ordering:
             >>> types.order(types.Uint(8), types.Uint(16))
             Ordering.LESS
 
+        Compare two :class:`Bool` types of differing const-ness::
+
+            >>> from qiskit.circuit.classical import types
+            >>> types.order(types.Bool(), types.Bool(const=True))
+            Ordering.GREATER
+
         Compare two types that have no ordering between them::
 
             >>> types.order(types.Uint(8), types.Bool())
             Ordering.NONE
+            >>> types.order(types.Uint(8), types.Uint(16, const=True))
+            Ordering.NONE
     """
     if (orderer := _ORDERERS.get((left.kind, right.kind))) is None:
         return Ordering.NONE
-    return orderer(left, right)
+    order_ = orderer(left, right)
+
+    # If the natural type ordering is equal (either one can represent both)
+    # but the types differ in const-ness, the non-const variant is greater.
+    # If one type is greater (and thus is the only type that can represent
+    # both) an ordering is only defined if that type is non-const or both
+    # types are const.
+    if left.const is True and right.const is False:
+        if order_ is Ordering.EQUAL:
+            return Ordering.LESS
+        if order_ is Ordering.GREATER:
+            return Ordering.NONE
+    if right.const is True and left.const is False:
```

Comment:
minor style: seeing `is True` and `is False` is slightly weird to me, given that they're regular booleans.  I'd just have expected `if left.const and not right.const` etc - actually I'm kind of surprised that pylint doesn't complain about it.

Reply:
Addressed in f1dc1a150ed8a1875bca3497cd225483461bd501.